packages/business-events/src/business-event-validator.ts

```diff
@@ -0,0 +1,27 @@
+import { BusinessEvent } from './business-event';
+
+export class BusinessEventValidator {
+  error: boolean = false;
+  message?: string;
+
+  constructor(public payload: BusinessEvent) {}
+
+  validate() {
+    this.validateEventContainsTopic();
+    this.validateEventContainsName();
+  }
+
+  validateEventContainsTopic() {
+    if (!this.error && !this.payload.topicArn) {
+      this.error = true;
+      this.message = 'Business event requires a topic ARN';
+    }
+  }
+
+  validateEventContainsName() {
+    if (!this.error && !this.payload.event) {
+      this.error = true;
+      this.message = 'Business event requires an event name';
+    }
+  }
+}
```

Comment:
It seems we might be able to remove the internal state of this class, by throwing errors inside the validation methods e.g.

```
export class BusinessEventValidator {
  constructor(private payload: BusinessEvent) {}

  validate(): void {
    this.validateEventContainsTopic();
    this.validateEventContainsName();
  }

  private validateEventContainsTopic(): void {
    if (!this.payload.topicArn) {
      throw new BusinessEventValidationError('Business event requires a topic ARN');
    }
  }

  private validateEventContainsName(): void {
    if (!this.payload.event) {
      throw new BusinessEventValidationError('Business event requires an event name');
    }
  }
}
``` 
(^ The access modifier change might not be necessary if we think we have benefit of calling the individual methods)

// We could create a dedicated class like so
```
class BusinessEventValidationError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'BusinessEventValidationError';
  }
}
```

The caller, in this instance `publish` then looks like

```
  static async publish(
    payload: BusinessEvent,
    config: Partial<SNSClientConfig> = BusinessEventPublisher.defaultConfig
  ): Promise<void> {
    try {
      const validator = new BusinessEventValidator(payload);

      validator.validate();

      await this.getClient(config).send(payload.command);
    } catch (error) {
      if (error instanceof BusinessEventValidationError) {
        console.error(`${BusinessEventValidator.name}: ${error.message}`);
      } else {
        console.error(`Failed to publish business event`, error);
      }
    }
  }
```

Reply:
Yeah I like this approach 👍